File: DSA/Big O/JS/02-test_performance.js
const my_array = ["one", "two", 'three', 'four', "five", 'six', "seven", "eight", "nine", "ten"]

function find_number(m_array){
    let start_time = performance.now();

    let x;
    for (x=0; x<m_array.length; x+=1){
        if (my_array[x] === "five"){
            console.log(`found ${my_array[x]}`)
        }
    }

    let end_time = performance.now();
    console.log(`find_number took time : ${end_time-start_time} milliseconds`)
}

find_number(my_array)

/* In JavaScript performance.now() method can be used to check the performance of your code.
You can check the execution time of your code using this method. */

// issue : can't run in my pc (may be node version is low)

File: DSA/Big O/Python/02-test_performance.py
import time
# The time module in Python provides functions for handling time-related tasks
# The time-related tasks includes :

# reading the current time
# formatting time
# sleeping for a specified number of seconds and so on.

my_list = ['one', 'two', 'three', 'four', 'five', 'six', 'seven', 'eight', 'nine', 'ten']

def findNumber(m_list):
    start_time = time.perf_counter()

    for m_list in my_list:
        if m_list == "ten":
            print(f"{my_list.index(m_list)} found {m_list}")

    end_time = time.perf_counter()

    print(f'findNumber function took time : {end_time-start_time} milliseconds')

findNumber(my_list)
